src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java

```diff
@@ -454,25 +484,48 @@ private AssemblyResult getAssemblyResult(final Haplotype refHaplotype, final int
-            initialSeqGraph.printGraph(new File(debugGraphOutputPath, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.1.initial_seqgraph.dot"), 10000);
-        }
+        // Either return an assembly result with a sequence graph or with an unchanged sequence graph deptending on the kmer duplication behavior
+        if (generateSeqGraph) {
+            final SeqGraph initialSeqGraph = rtgraph.toSequenceGraph();
+            if (debugGraphTransformations) {
+                rtgraph.printGraph(new File(debugGraphOutputPath, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.1.initial_seqgraph.dot"), 10000);
+            }
 
-        // if the unit tests don't want us to cleanup the graph, just return the raw sequence graph
-        if ( justReturnRawGraph ) {
-            return new AssemblyResult(AssemblyResult.Status.ASSEMBLED_SOME_VARIATION, initialSeqGraph, null);
-        }
+            // if the unit tests don't want us to cleanup the graph, just return the raw sequence graph
+            if (justReturnRawGraph) {
+                return new AssemblyResult(AssemblyResult.Status.ASSEMBLED_SOME_VARIATION, initialSeqGraph, null);
+            }
 
-        if ( debug ) {
-            logger.info("Using kmer size of " + rtgraph.getKmerSize() + " in read threading assembler");
-        }
-        printDebugGraphTransform(initialSeqGraph, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.2.initial_seqgraph.dot");
-        initialSeqGraph.cleanNonRefPaths(); // TODO -- I don't this is possible by construction
+            if (debug) {
+                logger.info("Using kmer size of " + rtgraph.getKmerSize() + " in read threading assembler");
+            }
+            printDebugGraphTransform(initialSeqGraph, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.2.initial_seqgraph.dot");
+            initialSeqGraph.cleanNonRefPaths(); // TODO -- I don't this is possible by construction
+
+            final AssemblyResult cleaned = cleanupSeqGraph(initialSeqGraph);
+            final AssemblyResult.Status status = cleaned.getStatus();
+            return new AssemblyResult(status, cleaned.getSeqGraph(), rtgraph);
+
+        } else {
+            if (debugGraphTransformations) {
+                rtgraph.printGraph(new File(debugGraphOutputPath, refHaplotype.getLocation() + "-sequenceGraph." + kmerSize + ".0.1.initial_seqgraph.dot"), 10000);
+            }
+
+            // if the unit tests don't want us to cleanup the graph, just return the raw sequence graph
+            if (justReturnRawGraph) {
+                return new AssemblyResult(AssemblyResult.Status.ASSEMBLED_SOME_VARIATION, null, rtgraph);
+            }
 
-        final AssemblyResult cleaned = cleanupSeqGraph(initialSeqGraph);
-        final AssemblyResult.Status status = cleaned.getStatus();
-        return new AssemblyResult(status, cleaned.getGraph(), rtgraph);
+            if (debug) {
```

Comment:
This code is common to the `if` and the `else` and can be moved up to before the conditional and un-duplicated.